fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java

```diff
@@ -2155,4 +2201,67 @@ private static void manageException(String message, boolean lenient) throws FHIR
+            // For example CareTeam.participant can be used to refer to a Patient or RelatedPerson resource:
+            // "participant": { "reference": "Patient/abc123" }
+            // "participant": { "reference": "RelatedPerson/abc456" }
+            for (Map.Entry<String, Set<String>> paramEntry : compartmentRefParams.entrySet()) {
+                final String searchParm = paramEntry.getKey();
+
+                // Ignore {def} which is used in the compartment definition where
+                // no other search parm is given (e.g. Encounter->Encounter).
+                if (!COMPARTMENT_PARM_DEF.equals(searchParm)) {
+                    SearchParameter sp = SearchUtil.getSearchParameter(resourceType, searchParm);
+                    if (sp != null && sp.getExpression() != null) {
+                        String expression = sp.getExpression().getValue();
+
+                        if (log.isLoggable(Level.FINE)) {
+                            log.fine("searchParam = [" + resourceType + "] '" + searchParm + "'; expression = '" + expression + "'");
+                        }
+                        Collection<FHIRPathNode> nodes = FHIRPathEvaluator.evaluator().evaluate(resourceContext, expression);
+                        for (FHIRPathNode node : nodes) {
+                            Reference reference = node.asElementNode().element().as(Reference.class);
+                            ReferenceValue rv = ReferenceUtil.createReferenceValueFrom(reference, baseUrl);
+                            if (rv.getType() != ReferenceType.DISPLAY_ONLY && rv.getType() != ReferenceType.INVALID) {
+                                // Check that the target resource type of the reference matches one of the
+                                // target resource types in the compartment definition.
+                                final String compartmentName = rv.getTargetResourceType();
+                                if (paramEntry.getValue().contains(compartmentName)) {
+                                    // Add this reference to the set of references we're collecting for each compartment
+                                    CompartmentReference cref = new CompartmentReference(searchParm, compartmentName, rv.getValue());
+                                    Set<CompartmentReference> references = result.computeIfAbsent(compartmentName, k -> new HashSet<>());
+                                    references.add(cref);
+                                }
+                            }
+                        }
+                    } else {
+                       log.warning("Compartment parameter not found: [" + resourceType + "] '" + searchParm + "'. This will stop compartment searches from working correctly.");
+                    }
+                }
+            }
+        } catch (Exception e) {
+            final String msg = "Unexpected exception extracting compartment references "
+                    + " for resource type " + resourceType;
```

Comment:
should wrap resourceType in single quotes

Reply:
fixed